element-hq/element-x-android
# Close the media preview screen ASAP with sending queue enabled

## Content

- When the media file is done processing and we've pressed the 'send' button, just dismiss the attachment preview instead of waiting for the file to be sent.
- For this, get the `@SessionCoroutineScope` from the `SessionMatrixModule` and use it to send the media even if the preview screen or the chat room screen are dismissed.
- As soon as the sending process begins in background, dismiss the attachment preview screen.

## Motivation and context

We decided to have this behaviour to improve the UX.

## Screenshots / GIFs

https://github.com/user-attachments/assets/a8461209-c7da-4830-ac21-8cad4ac16798

## Tests



- Enable the 'Media upload through send queue' developer option.
- Send a large media file, press the send button quickly.
- There may still be some progress dialog while the thumbnail is being generated and the media is compressed, if enabled.
- As seen in the video above, the preview screen should be dismissed and a local echo of the media file should appear.
- After a few seconds, the media should be marked as sent.

## Tested devices

- OS version(s): 14

## Checklist
